### PR TITLE
process creation uses new api

### PR DESCRIPTION
**Pull request by clavedeluna**
## Overview
*One more codemod, process creation, to rely on new API*

## Description

* minor changes that push us to generalization

## Documentation
* Does the README need to be updated?
* Do customer docs need to beg updated?
* Did you create any documentation related to this work?

## Describe your testing approach

* Happy/sad paths, Edge cases, Bad inputs
* Explanation of manual testing with any TS screenshots
* Or, an explanation of why this change does not need tests

## Performance

* How do you expect this PR to affect performance?

## Additional Details
* any follow up tickets or discussion
* Any specific merge / deploy details

### CODE REVIEW

File: codemodder/codemods/process_creation_sandbox.py [hunk 1]
```diff
@@ -1,62 +1,39 @@
...
-                    value=cst.Name(value=replacement_import),
-                    attr=cst.Name(value=get_call_name(original_node)),
-                ),
-                args=[cst.Arg(original_node.func), *original_node.args],
-            )
-            return new_call
-        return updated_node
+    def on_result_found(self, original_node, updated_node):
+        self.add_needed_import("security", "safe_command")
+        DependencyManager().add(["security==1.0.1"])
```
Comment (clavedeluna) [inline codemodder/codemods/process_creation_sandbox.py:33]: I decided to keep the DM API as is, instead of creating a helper wrapper method, since this is already an API. We can decide later on

**Review by andrecsilva — APPROVED** [review]

File: codemodder/codemods/process_creation_sandbox.py [hunk 2]
```diff
@@ -1,62 +1,39 @@
...
-            codemod_context,
-            self._results,
-            file_context.line_exclude,
-            file_context.line_include,
-        )
+    @classmethod
+    def rule(cls):
+        return """
+        rules:
+          - id: sandbox-process-creation
```
Comment (drdavella) [inline codemodder/codemods/process_creation_sandbox.py]: I think you mentioned this in a previous PR but I'm not sure why we need this field. Also the name is inconsistent with `NAME` above.

I think if you remove the `id` here and change `NAME` to `sandbox-process-creation` it should work.

**Review by drdavella — CHANGES_REQUESTED** [review]
Looks really nice overall. Just a minor change requested related to the rule ID/NAME field.

File: codemodder/codemods/process_creation_sandbox.py (hunk 2 above)
Comment (clavedeluna) [reply codemodder/codemods/process_creation_sandbox.py]: woops good catch on the incorrect name here! I was able to remove "id" here  by making the only thing under "rules" be `- pattern-either:`. limit-readline still needs the rule_id bc the yaml uses both `pattern-sources:` and `pattern-sinks:` . We will need to make the yaml generation more flexible.

**Review by drdavella — APPROVED** [review]
